frontend/src/components/game/ProblemPanel.tsx

```diff
@@ -0,0 +1,121 @@
+import MarkdownEditor from 'rich-markdown-editor';
+import { BottomFooterText, ProblemHeaderText, SmallText } from '../core/Text';
+import { DefaultButton, getDifficultyDisplayButton } from '../core/Button';
+import { Copyable } from '../special/CopyIndicator';
+import {
+  CenteredContainer,
+  FlexHorizontalContainer, FlexLeft, FlexRight, Panel,
+} from '../core/Container';
+import { Problem } from '../../api/Problem';
+import { NextIcon, PrevIcon } from '../core/Icon';
+
+const StyledMarkdownEditor = styled(MarkdownEditor)`
+  margin-top: 15px;
+  padding: 0;
+  
+  p {
+    font-family: ${({ theme }) => theme.font};
+  }
+
+  // The specific list of attributes to have dark text color.
+  .ProseMirror > p, blockquote, h1, h2, h3, ul, ol, table {
+    color: ${({ theme }) => theme.colors.text};
+  }
+`;
+
+const OverflowPanel = styled(Panel)`
+  overflow-y: auto;
+  height: 100%;
+  padding: 0 25px;
+`;
+
+const ProblemNavContainer = styled(FlexRight)`
+  align-items: baseline;
+  padding: 15px 0;
+`;
+
+const ProblemCountText = styled(SmallText)`
+  color: gray;
+`;
+
+type ProblemNavButtonProps = {
+  disabled: boolean,
+};
+
+const ProblemNavButton = styled(DefaultButton)<ProblemNavButtonProps>`
+  font-size: ${({ theme }) => theme.fontSize.default};
+  color: ${({ theme }) => theme.colors.gray};
+  background-color: ${({ theme, disabled }) => (disabled ? theme.colors.background : theme.colors.white)};
+  border-radius: 5px;
+  width: 35px;
+  height: 35px;
+  margin: 5px;
+  
+  box-shadow: 0 1px 6px rgba(0, 0, 0, 0.16);
+  
+  &:hover {
+    box-shadow: ${({ disabled }) => (disabled ? '0 1px 6px rgba(0, 0, 0, 0.16)' : '0 1px 6px rgba(0, 0, 0, 0.20)')};
+    cursor: ${({ disabled }) => (disabled ? 'default' : 'pointer')}; 
+  }
+  
+  i {
+    line-height: 35px;
+  }
+`;
+
+type ProblemPanelProps = {
+  problems: Problem[],
+  index: number,
+  onNext: (() => void) | null,
+  onPrev: (() => void) | null,
+};
+
+function ProblemPanel(props: ProblemPanelProps) {
+  const {
+    problems, index, onNext, onPrev,
+  } = props;
+
+  return (
+    <OverflowPanel>
+      <FlexHorizontalContainer>
+        <FlexLeft>
+          <div>
+            <ProblemHeaderText>{problems[index]?.name || 'Loading...'}</ProblemHeaderText>
+            {problems[index] ? getDifficultyDisplayButton(problems[index].difficulty) : null}
+          </div>
+        </FlexLeft>
+        <ProblemNavContainer>
+          <CenteredContainer>
+            <div>
+              <ProblemNavButton onClick={onPrev || undefined} disabled={!onPrev}>
+                <PrevIcon />
+              </ProblemNavButton>
+              <ProblemNavButton onClick={onNext || undefined} disabled={!onNext}>
+                <NextIcon />
+              </ProblemNavButton>
+            </div>
+            <ProblemCountText>
+              {`Problem ${index + 1} of ${problems.length}`}
+            </ProblemCountText>
+          </CenteredContainer>
+        </ProblemNavContainer>
```

Comment:
Hm yeah my main concern with this would be it causing a similar overlapping behavior as the console buttons, which can get weird if the screen size is small (or for the problem panel, if they simply scrolled down a little bit causing it to overlap with the description). Potentially we could add that row you mentioned to prevent it from overlapping, but I'm not sure if that's worth the design tradeoff/dev time to implement something like that - thoughts on leaving this for a post-MVP PR?

Reply:
Yes I think that's a valid concern - let's leave this for a potential post-MVP task!